### ISSUE TITLE
Modal focus management issues

### ISSUE THREAD

**Issue by cvrebert** [ResponsiveBP/Responsive#56]
* Clicking on a modal leaves the `<a>` that triggered the modal focused. For proper accessibility, the modal should become focused instead.
* Closing a modal returns focus to the `<body>`. For better accessibility, the focus should be returned to the link that originally activated the modal.

**Comment by JimBobSquarePants** [comment]
Thanks for that. In my tests the modal is focused on show. However you are right that the trigger should be focused on hide.

https://github.com/ResponsiveBP/Responsive/blob/v4/src/js/responsive.modal.js#L135

**Comment by cvrebert** [comment]
I beg to differ regarding show. Screencast on Mac Chrome:
![modal](https://cloud.githubusercontent.com/assets/419884/5655785/78baee98-9687-11e4-9f99-746b83113fd0.gif)

**Comment by JimBobSquarePants** [comment]
No worries, thanks . Feel free to submit a pull request if you spot what is wrong.

**Comment by cvrebert** [comment]
Not interested in investigating the cause... Anyway, happy hunting!
